### PR TITLE
[Merged by Bors] - feat: Actions on and by an order synonym

### PR DESCRIPTION
**Pull request by YaelDillies**
Complete the API about actions on and by `OrderDual`/`Lex`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Order/Group/Action/Synonym.lean [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+/-!
+# Actions by and on order synonyms
+-/
+
+variable {M N α : Type*}
+
+namespace OrderDual
+
+@[to_additive]
+instance instMulAction [Monoid M] [MulAction M α] : MulAction (Lex M) α := ‹MulAction M α›
```
Comment (eric-wieser) [inline Mathlib/Algebra/Order/Group/Action/Synonym.lean:18]: This is in the `OrderDual` namespace yet is about `Lex`.

File: Mathlib/Algebra/Order/GroupWithZero/Action/Synonym.lean [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+/-!
+# Actions by and on order synonyms
+-/
+
+variable {G₀ M₀ : Type*}
+
+namespace OrderDual
+
+instance instSMulWithZero [Zero G₀] [Zero M₀] [SMulWithZero G₀ M₀] : SMulWithZero G₀ᵒᵈ M₀ :=
```
Comment (eric-wieser) [inline Mathlib/Algebra/Order/GroupWithZero/Action/Synonym.lean:18]: Is the `SMul` instance defined in an earlier file? If so, can you mention where that instance is (and the lemmas around it) in the docstrings of the new files?

Comment (YaelDillies) [reply Mathlib/Algebra/Order/GroupWithZero/Action/Synonym.lean:18]: `Algebra.Order.Group.Action.Synonym` is where it is defined

File: Mathlib/Algebra/Order/Group/Action/Synonym.lean (hunk 1 above)
Comment (eric-wieser) [reply Mathlib/Algebra/Order/Group/Action/Synonym.lean:25]: This comment still applies to this line

**Review by b-mehta — APPROVED** [review]
bors merge

**Pull request closed without merging**